engine/src/main/java/org/terasology/rendering/logic/MeshRenderer.java

```diff
@@ -203,14 +202,13 @@ private void renderEntitiesByMaterial(SetMultimap<Material, EntityRef> meshByMat
                     location.getWorldPosition(worldPos);
                     float worldScale = location.getWorldScale();
 
-                    javax.vecmath.Matrix4f matrixWorldSpace = new javax.vecmath.Matrix4f(VecMath.to(worldRot), VecMath.to(worldPos), worldScale);
-                    transWorldSpace.set(matrixWorldSpace);
+                    Transform toWorldSpaceTransform = new Transform(worldPos, worldRot, worldScale);
 
-                    Vector3f worldPositionCameraSpace = new Vector3f();
-                    worldPositionCameraSpace.sub(worldPos, cameraPosition);
-                    Matrix4f matrixCameraSpace = new Matrix4f(worldRot, worldPositionCameraSpace, worldScale);
+                    Vector3f offsetFromCamera = new Vector3f();
+                    offsetFromCamera.sub(worldPos, cameraPosition);
+                    Matrix4f matrixCameraSpace = new Matrix4f(worldRot, offsetFromCamera, worldScale);
 
-                    AABB aabb = meshComp.mesh.getAABB().transform(transWorldSpace);
+                    AABB aabb = meshComp.mesh.getAABB().transform(toWorldSpaceTransform);
```

Comment:
One of my coding style aims is to write code that is as close as possible to plain English. In this case, I realise I OK'd the variable "toWorldSpaceTransform", but "toWorldSpace"  would have allowed to read this line as: "get the AABB of the mesh and transform it to world space". Right now it reads as "transform the transform". It's no big deal though and you can leave it. It's just an example of what I aim style-wise.

Reply:
I do understand what you are getting at and how it would be more readable in this case, but I feel that an even better option would be to move the method entirely into the `Transform` class as discussed above.

Reply:
I think I addressed this issue in another comment.

Reply:
So, to be clear, I think `toWorldSpaceTransform` should still become `toWorldSpace`.